spec/services/distribution_update_service_spec.rb

```diff
@@ -1,14 +1,48 @@
 RSpec.describe DistributionUpdateService, type: :service do
-  describe "call" do
-    # TODO: this function was extracted to the service object - do we need a parallel test?
-
-    let!(:distribution) { FactoryBot.create(:distribution, :with_items, item_quantity: 10) }
-    let!(:new_attributes) { { line_items_attributes: { "0": { item_id: distribution.line_items.first.item_id, quantity: 2 } } } }
+  subject { DistributionUpdateService }
+  let!(:storage_location) { create(:storage_location, :with_items) }
+  let!(:distribution) { FactoryBot.create(:distribution, :with_items, item_quantity: 10) }
+  let!(:new_attributes) { { line_items_attributes: { "0": { item_id: distribution.line_items.first.item_id, quantity: 2 } } } }
 
+  describe "call" do
     it "replaces a big distribution with a smaller one, resulting in increased stored quantities" do
       expect do
-        DistributionUpdateService.new(distribution, new_attributes).call
+        subject.new(distribution, new_attributes).call
       end.to change { distribution.storage_location.size }.by(8)
     end
+
+    it "returns a successful object with the distribution" do
+      result = subject.new(distribution, new_attributes).call
+      expect(result).to be_instance_of(subject)
+      expect(result).to be_success
+      expect(result.distribution).to be_valid
+    end
+
+    describe "#resend_notification?" do
+      it "evaluates to true if the issued at has changed" do
+        result = subject.new(distribution, { issued_at: distribution.issued_at + 1.week })
+        expect(result).to be_resend_notification
+      end
+    end
+
+    context "when there's not sufficient inventory" do
+      let(:too_much_params) { { organization_id: @organization.id, partner_id: @partner.id, storage_location_id: storage_location.id, line_items_attributes: { "0": { item_id: storage_location.items.first.id, quantity: 500 } } } }
```

Comment:
```suggestion
      let(:too_much_params) do 
           { 
               organization_id: @organization.id, 
               partner_id: @partner.id, 
               storage_location_id: storage_location.id, 
               line_items_attributes: { 
                 .... 
```
A bit of an optional stylistic thing for me. Could we use a `do/end` block to lay put the hash definition on different lines? I find it a little easier imo to see what the variable is that way.

Reply:
Yeah that's a good idea  -- it would make it easier to read :+1: